Implement concepts of fossil, biogenic, renewable and synthetic

l-emele opened the pull request:
Closes #108.

This PR implements the concepts of `fossil`, `biogenic`, `renewable` and `synthetic`.

akleinau commented:
"The origin is a quality describing the source." I dont really get a better understanding of the origin quality through this definition, it seems more like a synonym. Could you write a bit more?

The "origin" quality, by this definition, can be used for other things too that are not fuels. Like to state the origin of a dataset. Is this intended?

l-emele commented:
> "The origin is a quality describing the source." I dont really get a better understanding of the origin quality through this definition, it seems more like a synonym. Could you write a bit more?
> 
> The "origin" quality, by this definition, can be used for other things too that are not fuels. Like to state the origin of a dataset. Is this intended?

What about _"The origin **of a fuel** is a quality describing the source."_?

han-f commented:
Or how about: 
The origin indicates where something comes from (it’s source)?

stap-m commented:
> The origin indicates where something comes from (its source)?

The origin *is a quality that* indicates where something comes from (its source). ?

I think at the moment it is good to focus on fuels, but leaving the possibility open to increase this quality in the future.

l-emele commented:
It seems we reached a consensus so I changed the definition of `origin` to the formulation by @stap-m in my latest commit 4dacbb5.

Hannah already reviewed and approved this pull request. However as a) we changed the definition of origin and b) this is quite an extensive pull request changing a lot of lines I would be happy to have a second approval by either @stap-m and or @akleinau.

stap-m commented on oeo.omn:
```diff
@@ -3414,7 +3456,8 @@ Class: NonRenewableMunicipalWaste
     SubClassOf: 
-        MunicipalWaste
+        MunicipalWaste,
+        has_origin value fossil
```
here origin is also biogenic and renewable?

l-emele replied:
I intentionally applied no `has_origin` to `MunicipalWaste` as municipal waste per se has no _definied_ origin. Municipal waste has both an fossil and a biogenic/renewable fraction.

If we add `has_origin value fossil`, `has_origin value biogenic` and `has_origin value renewable` then this implies that municipal waste is fossil and biogenic/renewable at the same time.

If we do not add any `has_origin` to municipal waste than this implies that municipal waste has no defined origin. I wanted to express here, that only the subclasses `RenewableMunicipalWaste`and `NonRenewableMunicipalWaste` have defined origins. So not adding a `has_origin` to the classe `MunicipalWaste` is here in my view not a bug but a feature.

akleinau commented:
looks good
